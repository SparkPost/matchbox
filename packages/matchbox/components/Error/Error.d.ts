export default Error;
declare function Error(props: any): any;
declare namespace Error {
    const displayName: string;
}
