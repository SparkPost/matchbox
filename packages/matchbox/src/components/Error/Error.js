import React from 'react';
import { ErrorIcon } from '@sparkpost/matchbox-icons';
import { Box } from '../Box';

function Error(props) {
  const { className, error, wrapper: WrapperComponent = 'div' } = props;

  return (
    <Box as={WrapperComponent} className={className}>
      <Box as="span" display="inline-flex" color="red.700">
        <Box as="span" flex="0" mr="100" display="flex" alignItems="center">
          <ErrorIcon size={14} />
        </Box>
        <Box as="span" flex="1" fontSize="200" lineHeight="200">
          {error}
        </Box>
      </Box>
    </Box>
  );
}

Error.displayName = 'Error';
export default Error;
