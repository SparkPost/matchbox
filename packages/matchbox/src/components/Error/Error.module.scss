@import '../../styles/config';

.Error {
  padding-top: spacing(small);
}

.Icon {
  float: left;
  margin-right: rem(5);
  fill: color(red);
  line-height: rem(15);
}

.Message {
  display: inline-block;
  vertical-align: top;
  font-size: font-size(300);
  line-height: rem(15);
  font-weight: 500;
  color: color(red);
}
