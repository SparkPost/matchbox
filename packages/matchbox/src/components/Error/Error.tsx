import React from 'react';
import { ErrorIcon } from '@sparkpost/matchbox-icons';
import { Box } from '../Box';

type ErrorLabelProps = Pick<React.ComponentProps<typeof Box>, 'ml'> & {
  id?: string;
  className?: string;
  /**
   * @deprecated Use children instead
   */
  error?: React.ReactNode;
  children?: React.ReactNode;
  wrapper?: 'div' | 'span';
};

function ErrorLabel(props: ErrorLabelProps): JSX.Element {
  const { className, children, error, wrapper: WrapperComponent = 'div', id, ml = '0' } = props;

  return (
    <Box id={id} as={WrapperComponent} className={className} ml={ml} data-id="error-message">
      <Box
        as="span"
        color="red.700"
        fontSize="200"
        lineHeight="200"
        display="inline-flex"
        alignItems="flex-start"
      >
        <Box as="span" display="inline-block" mr="100" pt="2px" lineHeight="0">
          <ErrorIcon size={14} label="Error" />
        </Box>
        <span>
          {error}
          {children}
        </span>
      </Box>
    </Box>
  );
}

ErrorLabel.displayName = 'Error';
export default ErrorLabel;
