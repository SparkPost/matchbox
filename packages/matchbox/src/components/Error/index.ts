export { default as Error } from './Error';
export * from './Error';
