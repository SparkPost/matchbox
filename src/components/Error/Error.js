import React, { Component } from 'react';
import { Icon } from '../Icon';
import styles from './Error.module.scss';

const Error = ({ error }) => (
  <div className={styles.Error}>
    <Icon name='Error' className={styles.Icon} size={13} />
    <span className={styles.Message}>{ error }</span>
  </div>
);

export default Error;
