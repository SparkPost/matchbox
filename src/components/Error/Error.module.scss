@import '../../styles/config';

.Error {
  padding-top: spacing(small);
}

.Icon {
  fill: color(red, dark);
}

.Message {
  display: inline-block;
  padding-left: spacing(small);
  vertical-align: middle;
  font-size: rem(13);
  font-weight: 500;
  color: color(red, dark);
}
