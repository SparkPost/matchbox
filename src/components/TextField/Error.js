import React, { Component } from 'react';
import classnames from 'classnames';
import styles from './TextField.module.scss';

const Error = ({ error }) => (
  <div className={styles.Error}>
    { error }
  </div>
);

export default Error;
