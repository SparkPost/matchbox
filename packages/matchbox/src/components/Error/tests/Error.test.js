import React from 'react';
import Error from '../Error';
import { shallow } from 'enzyme';

describe('Error', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      error: 'Something terrible happened!'
    };

    wrapper = shallow(<Error {...props}/>);
  });

  it('renders error message', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
