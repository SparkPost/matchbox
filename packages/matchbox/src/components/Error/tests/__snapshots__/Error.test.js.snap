// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders error message 1`] = `
<div
  className="Error"
>
  <span
    className="Message"
  >
    <Error
      className="Icon"
      size={13}
    />
    Something terrible happened!
  </span>
</div>
`;
